pkg/webhook/v1beta1/pod/inject_webhook.go

```diff
@@ -144,6 +144,9 @@ func (s *SidecarInjector) Mutate(pod *v1.Pod, namespace string) (*v1.Pod, error)
 		return nil, err
 	}
 
+	// Add KATIB_TRIAL_NAME env name to all containers
+	mutatePodContainersEnv(mutatedPod, trial)
```

Comment:
If we mutate env vars to containers here, we can not mutate env vars to a metrics-collector container since we mutate a metrics-collector container to Pod in L173.

So should we move this process to place after all other mutations are done?

Reply:
To avoid refactoring all the Mutate function, I added manually the env when creating the metric collector container in getMetricsCollectorContainer.

Reply:
@axel7083 Should we put `mutatePodContainersEnv(mutatedPod, trial)` on L191?

Reply:
If doing so, we can remove L237-L240 maybe.

Reply:
I put the mutatePodContainersEnv at line [L148](https://github.com/kubeflow/katib/blob/efb18b376de261cd3f6336d506d5c16083041c33/pkg/webhook/v1beta1/pod/inject_webhook.go#L148) because the metric collector could be None, when it is none, the mutate return early:

https://github.com/kubeflow/katib/blob/efb18b376de261cd3f6336d506d5c16083041c33/pkg/webhook/v1beta1/pod/inject_webhook.go#L161-L163

And the goal of adding the mutatePodContainersEnv  and adding in getMetricsCollectorContainer is to avoid removing the early return, and limit the refractoring.
If desired, I can change to avoid making early return and place the mutatePodContainersEnv  at the end.

Reply:
Right. Thank you for the clarification.